### PR TITLE
created bookService online-book-store

### PR DESCRIPTION
**Pull request by OlgaStarikova**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <name>online-book-store</name>
     <description>online-book-store</description>
     <properties>
-        <java.version>21</java.version>
+        <java.version>20</java.version>
```
Comment (okuzan) [inline pom.xml:17]: why java 20? it's not LTS (long time support) release, I'd suggest switching to either 17 or 21 (preferably 17, it's in task descirption)

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1 +1,8 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/bookstoreDB?serverTimeZone=UTC
+spring.datasource.username=root
+spring.datasource.password=Sun%2002
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql=true
+
```
Comment (okuzan) [inline src/main/resources/application.properties:7]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

File: src/main/java/com/example/onlinebookstore/dto/BookDto.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
```
Comment (olekskov) [inline src/main/java/com/example/onlinebookstore/dto/BookDto.java:13]: It's not an entity, please remove annotations
```suggestion
public class BookDto {
    private Long id;
```

File: src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (olekskov) [inline src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java:12]: same
```suggestion
```

File: src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    private Long id;
```
Comment (olekskov) [inline src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java:13]: Id is assigned by DB, when you send request to create something you should not pass Id. User will obtain Id in response after creation
```suggestion
```

File: src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        EntityManagerFactory entityManagerFactory = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
```
Comment (olekskov) [inline src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java:62]: please add catch block

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+@Entity
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
```
Comment (okuzan) [inline src/main/java/com/example/onlinebookstore/model/Book.java:19]: these are critical fields, they shouldn't be nullable

**Review by okuzan — CHANGES_REQUESTED** [review]

**Pull request merged**